data/client/king_phisher/plugins/data_collection.py

```diff
@@ -0,0 +1,158 @@
+import argparse
+import copy
+import datetime
+import hashlib
+import sys
+
+from king_phisher import color
+from king_phisher import its
+from king_phisher import json_ex
+from king_phisher import version
+import king_phisher.client.plugins as plugins
+import king_phisher.client.gui_utilities as gui_utilities
+import king_phisher.client.widget.extras as extras
+
+from smoke_zephyr.utilities import unique
+
+__version__ = '1.0'
+
+def generate_hash(data, algorithm='sha256'):
+	if its.py_v3 and isinstance(data, str):
+		data = data.encode('utf-8')
+	return algorithm + ':' + hashlib.new(algorithm, data).hexdigest()
+
+def generate_signature(data):
+	data = copy.deepcopy(data)
+	data['hash'] = None
+	return generate_hash(json_ex.dumps(data))
+
+def verify_signature(data):
+	if not data.get('hash'):
+		return False
+	return data['hash'] == generate_signature(data)
+
+class Plugin(plugins.ClientPlugin):
+	authors = ['Spencer McIntyre', 'Erik Daguerre']
+	title = 'Anonymous Data Collection'
+	description = """
+	By default this plugin will be disabled, You must opt-in to this feature, to
+	have your campaign data anonymize, furthermore you will need to email the data in
+	for analyze.
+
+	By enabling this plugin, you are allowing this plugin to anonymize your campaign
+	data and prompt you to review the anonymous data collected and email it in for processing.
+	"""
+	homepage = 'http://github.com/securestate/king-phisher-plugins'
+	options = [
+		plugins.ClientOptionInteger('cycle_days', 'Number of days between reminders to submit data.', default=90,
+									display_name='Cycle in Days'),
+	]
+
+	def initialize(self):
+		version = __version__
+		self.signal_connect('server-connected', self.server_connected)
+
+		return True
+
+	def server_connected(self, _):
+		config = self.config
+		if self.config.get('last_date'):
+			if datetime.datetime.utcnow() - config['last_date'] < datetime.timedelta(days=config['cycle_days']):
+				return
+		dialog_txt = "It has been {} days since your last phishing data submission.\nWould you like to anonymize data for submission?".format(config['cycle_days'])
+		if not gui_utilities.show_dialog_yes_no('Phishing Data Collection', self.application.get_active_window(), dialog_txt):
+			return
+
+		get_stats = StatsGenerator(self.application.rpc)
+		stats = get_stats.generate_stats()
+
+		dialog = extras.FileChooserDialog('Save Anonymized Data', self.application.get_active_window())
+		file_name = 'anonmized_phishing_data.txt'
+		response = dialog.run_quick_save(file_name)
+		dialog.destroy()
+		if not response['target_path']:
+			return
+		file = open(response['target_path'], 'w')
+		file.write(stats)
+		gui_utilities.show_dialog_info('Please review and email:\n {} \nto KingPhisher@securestate.com'.format(response['target_path']),
```

Comment:
Also double quotes for format strings to be consistent with the rest of the project. It's handy to have them show up in a different color in the editor.